pkg/controller/controller_suite_test.go

```diff
@@ -62,3 +76,51 @@ var _ = AfterSuite(func() {
 	// Put the DefaultBindAddress back
 	metrics.DefaultBindAddress = ":8080"
 })
+
+var _ runtime.Object = &UnconventionalListType{}
+var _ runtime.Object = &UnconventionalListTypeList{}
+
+// UnconventionalListType is used to test CRDs with List types that
+// have a slice of pointers rather than a slice of literals.
+type UnconventionalListType struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              string `json:"spec,omitempty"`
+}
+
+// DeepCopyObject implements runtime.Object
```

Comment:
I handwrote these mostly because I was too lazy to figure out how to plug deepcopy-gen into a  hackscript and have it verified by CI. Also it may not be able to cope with a folder that contains multiple packages (`controller` and `controller_test`). Since the correctness of the DeepCopies is not really relevant for the test, thats IMHO good enough

Reply:
Can you add a comment with the above? :)